src/lib/utils/size_estimation_utils.hpp

```diff
@@ -0,0 +1,62 @@
+#pragma once
+
+#include <random>
+
+#include "types.hpp"
+
+namespace opossum {
+
+// In case the initially preallocated capacity is too small, the string is allocated on the heap.
```

Comment:
This comment does not really sound like method's general comment

Reply:
```
/**
  * This function iterates over a given vector of strings and sums up the memory usage of all elements. Due to the
  * small string optimization (SSO) in most current C++ libraries, each string has an initially allocated buffer (e.g.,
  * 15 chars in GCC's libstdc++). If a string is larger, the string is allocated on the heap and the initial string
  * object stores a pointer to the actual string on the heap.
  */
```

Better?